2019-03-Autumn/2_string_manipulation/custom_sort_string/custom_sort_string.cpp

```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+
+using namespace std;
+
+string customSortString(const string& S, const string& T) {
+    int charCounts[26];
+    fill(charCounts, charCounts + 26, 0);
+
+    // count char occurrences in T
+    for (const char& c: T) charCounts[c - 'a']++;
+
+    ostringstream resStream;
+    // append the chars present in S
+    for (const char& c: S)
+        while (charCounts[c - 'a']-- > 0)
+            resStream << c;
+    // append the chars for the rest
+    for (char c = 'a'; c <= 'z'; c++)
+        while (charCounts[c - 'a']-- > 0)
```

Comment:
I'm preparing to explain the postfix operation in the `while` loops...nice code golf (not necessarily the _most_ readable tho)